### ISSUE TITLE
`CA1508`: Null check + assignment throw the analyzer off

### ISSUE THREAD

**Issue by RenderMichael** [dotnet/roslyn-analyzers#6973]
### Analyzer

Diagnostic ID: [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): Avoid dead conditional code

### Analyzer source

Version: SDK 8.0 RC 1

### Describe the bug

```csharp
_ = (GetR() is { } a ? a : (a = null));

if (a is not null) // CA1508: 'a is not null' is always false
{
}

_ = a.X; // additional context: CS8602 is fired here

static R? GetR() => null;

record R(int X);
```

### Expected behavior

No warning - a is assigned null in some scenarios

### Actual behavior

Warning

## Additional context

if you use `a` without checking for null, you get a `CS8602`, so there's no winning.

**Comment by mavasani** [comment]
Verified this repros on latest bits
